### PR TITLE
Documentation for 1.4.0

### PR DESCRIPTION
**Pull request by ntonnaett**
That's just a start. We might want to add to this PR or have several PRs.

### CODE REVIEW

**Comment by ntonnaett** [comment]
Resolves #277

File: docs/Build/Meson_build.md [hunk 1]
```diff
@@ -1,36 +1,64 @@
...
+    ```bash
+    brew install meson qt rt-audio jack help2man
+    ```
+
+    Meson might not find qt when installed with brew. But brew tells you to set
+    PKG_CONFIG_PATH to a directory where it finds qt's pkgconfig file.
+    This directory has to be set as additional pkgconfig path in meson:
+
+    ```bash
+    meson --buildtype release --pkg-config-path PATH_GIVEN_BY_BREW build_release
```
Comment (dyfer) [inline docs/Build/Meson_build.md:31]: I think the pkg-config path can be obtained with `` `brew --prefix qt5`/lib/pkgconfig ``

File: docs/Build/Meson_build.md [hunk 2]
```diff
@@ -1,36 +1,64 @@
...
 ```
 
+Meson automatically downloads RtAudio and compiles it
+as a subproject, if RtAudio is not found on your system.
+If `help2man` is found, Meson will create a manpage from `jacktrip --help`.
+
 Now build with:
 ```bash
-cd builddir
+cd build_release
```
Comment (dyfer) [inline docs/Build/Meson_build.md:51]: Just making sure - are we building in a different directory on purpose?

File: docs/Documentation/MkDocs.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Write Documentation
+
+This documentation of JackTrip is generated with [Material](https://squidfunk.github.io/mkdocs-material/)
+for [MkDocs](https://www.mkdocs.org/). All pages are derived from Markdown files in the subdirectory `docs`
+in JackTrip's git repository. Setup and table of contents is found in a YAML file called mkdocs.yml in
+the root directory.
+
+If you only want to edit a page you can click on the pen symbol at the top of each page.
```
Comment (dyfer) [inline docs/Documentation/MkDocs.md:8]: Does that apply to locally served files or online?

**Review by dyfer — COMMENTED** [review]
Thanks! Looks good, just a few questions/comments.

File: docs/Build/Meson_build.md (hunk 2 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md:51]: Maybe it's enough to point out that you can create more than one build directory.

File: docs/Build/Meson_build.md (hunk 1 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md:31]: That's neat. Should I make one command out of it?

File: docs/Documentation/MkDocs.md (hunk 3 above)
Comment (ntonnaett) [reply docs/Documentation/MkDocs.md:8]: That's a feature that you can use when you are using MkDocs on services like GitHub, GitLab and maybe others. You can edit your pages directly on the Github website.

File: docs/Build/Meson_build.md (hunk 1 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md]: Do you want to test the changed command? I don't have MacOS.

File: docs/About/License.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# License
+
+The code for the graphical user interface portion is released under version 3.0 of the GPL.
+So every build of JackTrip with a graphical user interface is only licensed as GPL.
+With the introduction of GPL licensed dependencies in the future more parts of JackTrip
+wil become GPL. 
+
+Only files that were part of the original JackTrip project are covered by the more permissive license in LICENSE\_nogui.
+
+--8<-- "LICENSE.md"
```
Comment (dyfer) [inline docs/About/License.md:10]: is this a valid markdown?

File: docs/Build/Meson_build.md [hunk 5]
```diff
@@ -1,29 +1,61 @@
...
+=== "Debian/Ubuntu"
+
+    ```bash
+    apt install meson build-essential qtbase5-dev librtaudio-dev libjack-jackd2-dev help2man
+    ```
+
+=== "MacOS"
+
+    ```bash
+    brew install meson qt rt-audio jack help2man
```
Comment (dyfer) [inline docs/Build/Meson_build.md:23]: sorry, just noticed this. `qt` in homebrew is now Qt 6. Do we support that? If not, then I'd change this to `qt5`

File: docs/Build/Meson_build.md [hunk 6]
```diff
@@ -1,29 +1,61 @@
...
+    ```
+
+    Meson might not find qt when installed with brew. But brew tells you to set
```
Comment (dyfer) [inline docs/Build/Meson_build.md:26]: Just a suggestion - I think the core of the issue is that qt is by default not symlinked to `/usr/local` for some presumably valid reasons. I'd guess that that's the reason meson is not finding it? If so, maybe the following would be more accurate:

`Qt installed with brew is not symlinked to`/usr/local` by default. Brew tells you (...)`

**Review by dyfer — CHANGES_REQUESTED** [review]
Thanks!

I have some minor suggestions. Also:
- do we want an empty "Authors" file?
- was the change to `LICENSE.md` for QJackTrip and `LICENSE_nogui.md` for JackTrip part agreed to by everyone? I can't remember if we've talked about it at the last meeting.

File: docs/About/License.md (hunk 4 above)
Comment (ntonnaett) [reply docs/About/License.md:10]: You can add markdown extensions in mkdocs.yml. This one is for including files.

File: docs/Build/Meson_build.md (hunk 5 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md:23]: I will change this.

File: docs/Build/Meson_build.md (hunk 6 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md:26]: If qt doesn't want to live in `/usr/local` it should append its pkgconfig directory to PKG_CONFIG_PATH. So your first suggestion looks fine to me. Could you test it?

Comment (dyfer) [reply docs/Build/Meson_build.md:26]: Sorry if I wasn't clear - I'm not contradicting my previous suggestion (I think), but rather wanted to clarify the reason for not finding qt - it's not an issue with meson, or qt install, but rather brew's decision to not link it by default. OTOH it's fine to leave your comment as is, totally up to you.

**Comment by ntonnaett** [comment]
>     * do we want an empty "Authors" file?

No, actually this PR should only be a starting point. But I can add authors. Some copyright statements in the project are misleading too.
 
>     * was the change to `LICENSE.md` for QJackTrip and `LICENSE_nogui.md` for JackTrip part agreed to by everyone? I can't remember if we've talked about it at the last meeting.

I understood that we wanted to have GPL as an umbrella license and see MIT only as a possbility for single files. When we add weakjack the nogui build becomes GPL too.

**Comment by dyfer** [comment]
> When we add weakjack the nogui build becomes GPL too.

Really? Even when we don't actually build with weakjack? I think I'm still unclear on this.

We were not planning on distributing nogui build with weakjack - weakjack would be used on macos and windows builds only, which do include the gui.

**Comment by dyfer** [comment]
Also, in general things look good to me. I'd prefer if we took a look maybe as a group at the dev meeting to make sure everyone is on board with the changes etc.

File: docs/Build/Meson_build.md (hunk 6 above)
Comment (ntonnaett) [reply docs/Build/Meson_build.md:26]: It's kind of strange. Because other libraries work out of the box.

**Comment by ntonnaett** [comment]
> > When we add weakjack the nogui build becomes GPL too.
> 
> Really? Even when we don't actually build with weakjack? I think I'm still unclear on this.
> 
> We were not planning on distributing nogui build with weakjack - weakjack would be used on macos and windows builds only, which do include the gui.

I would think so. You usually set a macro variable that sets if you want weakjack or not. But it's handled inside weakjack. All jack includes are replaced by one weakjack include.

Personally I'm not quite sure for whom we try to save that MIT build. I don't really think anybody cares.

**Comment by dyfer** [comment]
> I would think so. You usually set a macro variable that sets if you want weakjack or not. But it's handled inside weakjack. All jack includes are replaced by one weakjack include.

Ah I see. I previously thought that we'd use `#ifdef` for including weakjack or not, but I now see that this is not how you'd normally use it. We can re-visit that when we add weakjack (which I'm planning to do after working on the GHA build script installers etc PR)

**Comment by ntonnaett** [comment]
> Also, in general things look good to me. I'd prefer if we took a look maybe as a group at the dev meeting to make sure everyone is on board with the changes etc.

That's a good idea. Especially for the contributors page.

File: docs/About/Contributors.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+??? note "Aaron Wyatt /// ***Monash University***"
+
+    **2019 - **
+
+    Feature Work
+
+    * Graphical User Interface (QJackTrip)
+    * Authentication
+    * Installer Creation Scripts
```
Comment (psiborg112) [inline docs/About/Contributors.md:24]: Can we add "IPv6 support" to my feature list?

Comment (ntonnaett) [reply docs/About/Contributors.md:25]: Done

File: docs/Build/Meson_build.md (hunk 1 above)
Comment (dyfer) [reply docs/Build/Meson_build.md]: Tested on macOS with the updated instructions - everything works as expected

**Review by dyfer — APPROVED** [review]
Thanks! All my questions have been addressed.

Can't remember if this PR is waiting on some changes after the last meeting - let's have @ntonnaett merge it once ready.

**Comment by ntonnaett** [comment]
If we merge this, this will immediatley become our documentation. Do we want this now?

**Pull request merged**